lua/entities/gmod_wire_expression2/core/custom/streamcore.lua

```diff
@@ -5,24 +5,77 @@ CreateConVar( "streamcore_antispam_ignoreadmins", 1, FCVAR_SERVER_CAN_EXECUTE )
 CreateConVar( "streamcore_adminonly", 0, FCVAR_SERVER_CAN_EXECUTE )
 CreateConVar( "streamcore_maxradius", 120, FCVAR_SERVER_CAN_EXECUTE )
 
-util.AddNetworkString( "XTS_SC_StreamStart" )
-util.AddNetworkString( "XTS_SC_StreamStop" )
-util.AddNetworkString( "XTS_SC_StreamVolume" )
-util.AddNetworkString( "XTS_SC_StreamRadius" )
-util.AddNetworkString( "XTS_SC_StreamHelp" )
+util.AddNetworkString( "CFC_SC_StreamStart" )
+util.AddNetworkString( "CFC_SC_StreamStop" )
+util.AddNetworkString( "CFC_SC_StreamVolume" )
+util.AddNetworkString( "CFC_SC_StreamRadius" )
+util.AddNetworkString( "CFC_SC_StreamHelp" )
+
+local rawget = rawget
+local CurTime = CurTime
+local clamp = math.Clamp
+
+local stringLen = string.len
+local stringSub = string.sub
+local stringTrim = string.Trimp
 
 local streams = {}
 local antispam = {}
+local urlFixes = {}
+
+local throttleConfig = {
+    delays = { -- In seconds, minimum delay between uses
+        default = 0.1,
+        streamRadius = 0.25,
+        streamVolume = 0.25,
+        streamStop = 0.25
+    }
+}
+
+local function setLastUse( chip, funcName )
+    local ent = chip.entity
+
+    local throttles = ent.streamcoreThrottle
+    if not throttles then ent.streamcoreThrottle = {} end
```

Comment:
gotta assign throttles to this new table else it'll still be nil when you set it to CurTime
```suggestion
    if not throttles then
        throttles = {}
        ent.streamcoreThrottle = throttles
    end
```